core/spec/models/spree/order_spec.rb

```diff
@@ -1600,6 +1600,40 @@ def generate
     it 'creates at least one new shipment for the order' do
       expect { subject }.to change { order.shipments.count }.by 1
     end
+
+    context "with a custom inventory unit builder" do
+      before do
+        class TestInventoryUnitBuilder
+          def initialize(order)
+          end
+
+          def missing_units_for_line_item(line_item)
+            []
+          end
+        end
+
+        # This return value is stubbed this way, so that it can access the
+        # arbitrary_inventory_unit let variable.
+        allow_any_instance_of(TestInventoryUnitBuilder)
+          .to receive(:missing_units_for_line_item)
+          .and_return([arbitrary_inventory_unit])
```

Comment:
I think you can work around this issue if you define the class like the following:

```ruby
TestInventoryUnitBuilder = Class.new do
  def initialize(order)
  end

  define_method(:missing_units_for_line_item) do |_line_item|
    [arbitrary_inventory_unit]
  end
end
```

Reply:
I thought I tried that and it didn't work... but let me double check that.

Reply:
Ah, I just needed to capture it in a local variable.